include/TPP/Transforms/Utils/VNNIUtils.h

```diff
@@ -40,12 +45,19 @@ std::optional<int64_t> getVnniBlockingFactor(Type type);
 // Return true if the memref is in VNNI layout with rank `expectedRank`.
 bool isInVnniLayout(VnniOperandRank expectedRank, MemRefType memref);
 
+// Return true if the memref is in VNNI layout with rank `expectedRank`.
+bool isInVnniLayout(VnniOperandRank expectedRank, VectorType vector);
```

Comment:
The we need VNNI Utils for transpose. I would suggest adding Norm->VNNI function call lowering in a different PR to keep PRs small.

Reply:
We need this for when contraction inputs are not compatible to be lowered into brgemm. Should we seperate that out into a different patch? If yes, I'll move this out and keep just the transpose lowering in this patch for now. That makes the pr quite clean and small.

Reply:
> If yes, I'll move this out and keep just the transpose lowering in this patch for now.

Yes, please. Keep here _only_ transpose logic.